kart/point_cloud/import_.py

```diff
@@ -332,32 +280,75 @@ def _convert_tile_to_copc(source, dest):
     assert dest.is_file()
 
 
-class ListBasedSet:
-    """
-    A basic set that doesn't use hashing, so it can contain dicts.
-    Very inefficient for lots of elements, perfect for one or two elements.
-    """
-
-    def __init__(self):
-        self.list = []
+def _unwrap_metadata(metadata):
+    if isinstance(metadata, str):
+        metadata = json.loads(metadata)
+    if "metadata" in metadata:
+        metadata = metadata["metadata"]
+    return metadata
 
-    def add(self, element):
-        if element not in self.list:
-            self.list.append(element)
 
-    def only(self):
-        """Return the only element in this collection, or raise a LookupError."""
-        if len(self.list) != 1:
-            raise LookupError(
-                f"Can't return only element: set contains {len(self.list)} elements"
-            )
-        return self.list[0]
+def extract_pc_tile_metadata(pc_tile_path, extract_schema=False):
+    """
+    Use pdal to get any and all point-cloud metadata we can make use of in Kart.
+    This can include metadata must be dataset-homogenous and would be stored in the dataset's /meta/ folder,
+    along with other metadata that is tile-specific and would be stored in the tile's pointer file.
+    """
+    import pdal
 
-    def __contains__(self, element):
-        return element in self.list
+    config = [
+        {
+            "type": "readers.las",
+            "filename": str(pc_tile_path),
+            "count": 0,  # Don't read any individual points.
+        }
+    ]
+    if extract_schema:
+        config.append({"type": "filters.info"})
 
-    def __len__(self):
-        return len(self.list)
+    pipeline = pdal.Pipeline(json.dumps(config))
+    try:
+        pipeline.execute()
+    except RuntimeError:
+        raise InvalidOperation(
+            f"Error reading {pc_tile_path}", exit_code=INVALID_FILE_FORMAT
+        )
 
-    def __iter__(self):
-        return iter(self.list)
+    metadata = _unwrap_metadata(pipeline.metadata)
+    info = metadata["readers.las"]
+
+    result = {
+        "compressed": info["compressed"],
+        "version": f"{info['major_version']}.{info['minor_version']}",
+        "copc-version": get_copc_version(info),
+        "point-data-record-format": info["dataformat_id"],
+        "point-data-record-length": info["point_length"],
+        "crs": info["srs"]["wkt"],
+        "native-extent": get_native_extent(info),
+        "count": info["count"],
+    }
+    if extract_schema:
+        result["schema"] = metadata["filters.info"]["schema"]
+    return result
+
+
+def pc_tile_metadata_to_pointer_metadata(metadata):
+    """
+    Given all the tile-metadata, returns thats which should be written to an LFS pointer file,
+    in the appropriate format.
+    """
+    return {
+        # TODO - available.<URL-IDX> <URL>
+        # TODO - find the real extent.crs84
+        "extent.crs84": _format_array(metadata["native-extent"]),
```

Comment:
should probably just not return `extent.crs84` until implemented?

`available` is a bit orthogonal to point-clouds, not sure whether we handle it somewhere else?

Reply:
Removed extent.crs84, removed the available TODO